lib/ecto/query/builder/from.ex

```diff
@@ -84,26 +89,28 @@ defmodule Ecto.Query.Builder.From do
 
   defp query(prefix, source, schema, as) do
     aliases = if as, do: [{as, 0}], else: []
+    from_fields = [source: {source, schema}, as: as, prefix: prefix]
+
+    query_fields = [
+      from: {:%, [], [Ecto.Query.FromExpr, {:%{}, [], from_fields}]},
+      aliases: {:%{}, [], aliases}
+    ]
 
-    {:%, [], [Ecto.Query,
-              {:%{}, [],
-               [from: {:%, [], [Ecto.Query.FromExpr,
-                                {:%{}, [], [source: {source, schema}, as: as]}]},
-                prefix: prefix,
-                aliases: {:%{}, [], aliases}]}]}
```

Comment:
User backwards incompatible!!! We may need to keep the source prefix and from prefix separated.